plugins/enhanced_titles/__init__.py

```diff
@@ -0,0 +1,429 @@
+                sortfield = field + "sort"
+                metadata[sortfield] = sortname
+            else:
+                self._swapprefix(metadata, field)
+
+    def _swapprefix(self, metadata, field):
+        """Swaps the prefix of the title based on the album/track language."
+
+        If no language information is found, then it uses all languages available.
+        Otherwise, if none of the languages are available, it just copies the title.
+        Otherwise it uses exclusively the languages that are also available.
+        """
+        sortfield = field + "sort"
+        languages = [metadata["language"], metadata["_releaselanguage"]]
+        languages = [language for language in languages if language]
+        if not languages:
+            prefixes = _articles[""] + [article.capitalize() for article in _articles[""]]
+            metadata[sortfield] = func_swapprefix(None, metadata[field], *prefixes)
+        else:
+            languages = [language for language in languages if language in _articles.keys()]
+            if not languages:
+                metadata[sortfield] = metadata[field]
+            else:
+                prefixes = lang_functions.find_prefixes(languages)
+                metadata[sortfield] = func_swapprefix(None, metadata[field], *prefixes)
+
+    def set_track_titlesort(self, album, metadata, track, release):
+        """Sets the track's titlesort field.
+        """
+        if config.setting[ENABLE_TAGGING]:
+            handler = partial(self._response_handler, metadata = metadata, field = "title")
+            if config.setting[CHECK_TRACK]:
+                MBAPIHelper(album.tagger.webservice).get_track_by_id(
+                    metadata["musicbrainz_recordingid"],
+                    handler,
+                    inc = ["aliases"]
+                )
+            else:
+                self._swapprefix(metadata, "title")
+
+    def set_album_titlesort(self, album, metadata, release):
+        """Sets the album's albumsort field.
+        """
+        if config.setting[ENABLE_TAGGING]:
+            handler = partial(self._response_handler, metadata = metadata, field = "album")
+            if config.setting[CHECK_ALBUM]:
+                ReleaseGroupHelper(album.tagger.webservice).get_release_group_by_id(
+                    metadata["musicbrainz_releasegroupid"],
+                    handler,
+                    inc = ["aliases"]
+                )
+            else:
+                self._swapprefix(metadata, "album")
+
+
+class LangFunctions:
+    """Provides scripting functions swapprefix_lang, delprefix_lang, and title_lang.
+
+    Combinations of languages and their prefixes/minor words are stored after
+    they're used at least once.
+    """
+
+    prefixes_lists = {}
+    titles_lists = {}
+
+    def _format_languages(self, languages):
+        languages = [lang.lower()[:3] for lang in languages]
+        languages = [lang for lang in languages if lang in _articles.keys()]
```

Comment:
Iterator for dict defaults to keys:

```suggestion
        languages = [lang for lang in languages if lang in _articles]
```

Reply:
Perhaps it could be done differently (it's prolly faster) using `set.intersection`:

```python
langs = set(lang[:3].lower() for lang in languages)
articles_langs = set(_articles)
return tuple(langs.intersection(articles_langs))
```

Also better truncating and apply `lower()` on 3 letters, than lowering the full string and truncating to 3 letters.

Reply:
I would suggest that `_articles_langs = set(_articles)` would be better off done once immediately after `_articles[""]` is defined.